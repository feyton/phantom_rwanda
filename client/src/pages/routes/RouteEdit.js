import React from 'react';
import { useLocation, useParams } from 'react-router-dom';
import { toast } from 'react-toastify';
import { axiosBase as axios } from '../../utils/Api.js';
import RouteComponentEdit from './components/RouteComponentEdit.js';

const RouteEdit = () => {
  const location = useLocation();
  const routeInfo = location.state?.route;
  const { id } = useParams();
  const handleEdit = async (data) => {
    try {
      await axios.put(`/routes/${id}`, data);
      toast('Route updated successfully', { type: 'success' });
      return;
    } catch (error) {
      return Promise.reject(error);
    }
  };

  return (
    <RouteComponentEdit
      formAction={handleEdit}
      formTitle="Update Route"
      instance={routeInfo}
    />
  );
};

export default RouteEdit;
