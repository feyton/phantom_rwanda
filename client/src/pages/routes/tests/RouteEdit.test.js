import { mount } from 'enzyme';
import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import RouteEdit from '../RouteEdit.js';

const route = {
  origin: 'Kimironko',
  destination: 'Down-town',
  distance: 30
};

const mockUseLocationValue = {
  pathname: '/testroute',
  search: '',
  hash: '',
  state: { route }
};
jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useLocation: jest.fn().mockImplementation(() => {
    return mockUseLocationValue;
  })
}));

describe('Edit Page', () => {
  it('should render the Edit component', () => {
    const elem = renderer
      .create(
        <MemoryRouter>
          <RouteEdit />
        </MemoryRouter>
      )
      .toJSON();
    expect(elem).toMatchSnapshot();
  });

  it('should render the Route component', () => {
    const wrapper = mount(
      <MemoryRouter>
        <RouteEdit />
      </MemoryRouter>
    );
    expect(wrapper.find('RouteComponent').length).toEqual(1);
  });
});
