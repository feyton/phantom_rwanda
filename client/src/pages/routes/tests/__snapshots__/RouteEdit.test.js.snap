// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Edit Page should render the Edit component 1`] = `
<div
  className="w-full"
>
  <form
    className="p-8 shadow-main px-4 flex flex-col w-full md:max-w-lg md:mx-auto"
    onSubmit={[Function]}
  >
    <h1
      className="font-bold text-xl pb-8 text-center "
    >
       
      Update Route
       
    </h1>
    <div
      className="block mb-2"
    >
      <label
        className="block text-grey-darker text-sm font-bold mb-2"
        htmlFor="origin"
      >
        Origin:
      </label>
      <input
        className="appearance-none border font-raleway rounded w-full py-2 px-3 text-grey-darker bg-gray-200 text-md outline-hidden"
        defaultValue="Kimironko"
        name="origin"
        required={true}
      />
    </div>
    <div>
      <label
        className="block text-grey-darker text-sm font-bold mb-2"
        htmlFor="destination"
        name="distance"
      >
        Destination
      </label>
      <input
        className="appearance-none border font-raleway rounded w-full py-2 px-3 text-grey-darker bg-gray-200 text-md outline-hidden"
        defaultValue="Down-town"
        name="destination"
        required={true}
        type="text"
      />
    </div>
    <div>
      <label
        className="block text-grey-darker text-sm font-bold mb-2"
        htmlFor="distance"
        name="distance"
      >
        Distance
      </label>
      <input
        className="appearance-none border font-raleway rounded w-full py-2 px-3 text-grey-darker bg-gray-200 text-md outline-hidden"
        defaultValue={30}
        name="distance"
        required={true}
        type="number"
      />
    </div>
    <div
      className="self-center mt-6"
    >
      <div>
        <button
          className="bg-primary text-sm text-white py-2 px-3 hover:bg-hover rounded-xl py-2 text-base md:text-sm px-5"
          type="submit"
        >
          Update Route
        </button>
      </div>
    </div>
  </form>
</div>
`;
